bigquery/storage/managedwriter/managed_stream.go

```diff
@@ -324,6 +328,8 @@ func (ms *ManagedStream) lockingAppend(pw *pendingWrite) error {
 	}
 	if err != nil {
 		if shouldReconnect(err) {
```

Comment:
We should reconnect if we experience over x times of in stream retries. And also there are some known internal errors that needs to be treated as unavailable:
https://screenshot.googleplex.com/9xtULSCaXLMJSJT

Reply:
aborted/cancelled/unavailable are all valid codes for retry scenarios already.

The notion of in-connection errors is not encoded in this PR.  Can you clarify?  Do you mean X sequential errors, or X total on a given connection? Is this error restricted to certain status codes?  Are the errors from the bidi receive, or are they embedded in the response message?

Reply:
I mean if you have already retried X times on the same connection without succeeding, maybe it is worth to reconnect. We had some issues when the whole connection and not able to recover, it should be a bug on our side but it would be nice to cover it up in the client. (I failed to find the bug... the bug was on internal error).

By establishing a new connection, we found a new task and retry everything. So it is worth doing that.

Reply:
I'm inclined to put this logic in a subsequent PR to prevent this one becoming too unwieldy.  Added the scenario to the existing design doc.

Reply:
Sure.